r/R/table.R

```diff
@@ -331,12 +331,5 @@ as_arrow_table.arrow_dplyr_query <- function(x, ...) {
   # See query-engine.R for ExecPlan/Nodes
```

Comment:
Can all of this be simplified now to `as_arrow_table(as_record_batch_reader(x))` now?